File: client/src/modules/inventory/types/roomEquipment.ts
import type { Room } from '@/modules/room/types/room';
import type { InventoryType } from './inventory';
import * as yup from 'yup';
export const roomEquipmentSchema = yup.object({
  roomID: yup.number().required('Room is required').min(1, 'Please select a room'),
  quantity: yup
    .number()
    .transform((value, originalValue) =>
      originalValue === '' || Number.isNaN(value) ? undefined : value,
    )
    .required('Quantity is required')
    .min(0, 'Quantity must be at least 0')
    .max(100, 'Quantity must be at most 100'),
});

export type RoomEquipmentFormType = yup.InferType<typeof roomEquipmentSchema>;

export type RoomEquipmentType = {
  roomEquipmentID: number;
  roomID: number;
  inventoryID: number;
  quantity: number;
  isAvailable: boolean;
};

File: server/Backend/Models/RoomEquipment.cs
namespace Backend.Models;

public class RoomEquipment
{
    public int RoomEquipmentID { get; set; }
    public int RoomID { get; set; }
    public int InventoryID { get; set; }
    public int Quantity { get; set; }
    // Navigation
    public Room Room { get; set; }
    public Inventory Inventory { get; set; }
}